### ISSUE TITLE
convert: Unable to write sRGB png

### ISSUE THREAD

**Issue by heitorPB** [ImageMagick/ImageMagick#2504]
### Prerequisites

- [x] I have written a descriptive issue title
- [x] I have verified that I am using the latest version of ImageMagick
- [x] I have searched [open](https://github.com/ImageMagick/ImageMagick/issues) and [closed](https://github.com/ImageMagick/ImageMagick/issues?q=is%3Aissue+is%3Aclosed) issues to ensure it has not already been reported

### Description

I used to convert pdfs to pngs like `convert in.pdf out.png` and it would create a coloured png. But now it only creates a grayscale png.

### Steps to Reproduce

```
$ convert variances.pdf variances.png
convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `variances.png' @ warning/png.c/MagickPNGWarningHandler/1748.
$ identify variances.p*
variances.pdf PDF 422x311 422x311+0+0 16-bit sRGB 26430B 0.000u 0:00.000
variances.png PNG 422x311 422x311+0+0 8-bit Grayscale Gray 19120B 0.000u 0:00.000
```

Adding `-colorspace sRGB` makes no difference in the out png.

But `$ convert -density 300 variances.pdf variances.png` produces a bigger image with every symbol half-black and half-coloured.

`$ convert -density 600 variances.pdf variances.png` produces an even bigger image, with all the colours of the pdf.

Something changed in the last ~2 months, i did not need to specify the `density` previously.

### System Configuration

- ImageMagick version: 7.0.10.28-1
- libpng version: 1.6.37-3
- ghostscript version: 9.52-1
- Environment (Operating system, version and so on): ArchLinux

**Comment by fmw42** [comment]
Post a PDF that has this issue. What is your Ghostscript version? If old, please update. There were a number of versions of GS that had issues.

**Comment by heitorPB** [comment]
I amended the first post to include the postscript version. And it is 9.52-1.

Here's a minimum python script to generate a problematic pdf with colours:

```python
import numpy as np
import matplotlib.pyplot as plt

x = np.arange(0.1, 4, 0.2)
y = np.exp(-x)

fig, ax = plt.subplots()
m, c, b = ax.errorbar(x, y,
                      fmt='o',
                      color='blue'
                      )
m.set_markeredgecolor('black') # add border to points

fig.savefig('error.pdf', bbox_inches='tight')
```
Python 3.8, matplotlib 3.3.1-1

And the pdf: [error.pdf](https://github.com/ImageMagick/ImageMagick/files/5170212/error.pdf)

Using `convert` on this pdf:

```
$ convert error.pdf error.png
convert: profile 'icc': 'RGB ': RGB color space not permitted on grayscale PNG `error.png' @ warning/png.c/MagickPNGWarningHandler/1748.
```

I was digging this code and noticed that if I remove the line `m.set_markeredgecolor('black')`, ImageMagick can create a coloured png. Here's the good pdf: [working.pdf](https://github.com/ImageMagick/ImageMagick/files/5170228/working.pdf)

```
$ file error.pdf working.pdf
error.pdf:   PDF document, version 1.4
working.pdf: PDF document, version 1.4

$ identify error.pdf working.pdf
error.pdf PDF 395x297 395x297+0+0 16-bit sRGB 9027B 0.000u 0:00.000
working.pdf PDF 395x297 395x297+0+0 16-bit sRGB 8747B 0.000u 0:00.000
```

**Comment by fmw42** [comment]
So you have resolved the issue. It is a faulty PDF created by matplotlib

**Comment by heitorPB** [comment]
Both PDFs are valid PDFs.

I'm not an expert in PDFs, so I can't say it is a "faulty PDF". Do you know how to investigate this further?

**Comment by fmw42** [comment]
I do not know. It is "faulty", in my opinion, in that you specify a parameter that is causing the pdf to be read as grayscale. But an IM developer should comment further on this issue.

**Comment by snibgo** [comment]
Your OP was about variances.pdf, but I can't see a link to it.

You do link to error.pdf. This is mostly black and white, with small blue spots. If this is rasterised with a low density, then the blue dots are smothered by the black edges and vanish, and the result is monochrome.

error.pdf doesn't have an ICC profile, and I don't get a warning message about that, with v7.0.10-27 and Ghostscript 9.19 on Windows 8.1.

I can't see a fault in the PDF, nor any problem.

**Comment by heitorPB** [comment]
Downgrading to ghostscript-9.50-2 removed the warning/error from the convert command and produced an sRGB png. Ghostscript 9.51-1 and newer produce grayscale pngs, with this example.

With newer Ghostscripts, I need to increase the density of the pdf and scale down the output to get a similar result. With gs 9.52-2, this works:
```
$ convert -density 600 error.pdf -resize 12% error.png
```
The resize to 12% is to have the same size as without the density of 600.

**Comment by urban-warrior** [comment]
Post this issue to the Ghostscript developers.  With 9.52-2, a density of 100 or 200 return fully black markers.  Try 300 and you get half black and half blue markers.  With 600 you can full blue markers.

**Comment by heitorPB** [comment]
I posted this in the Ghostscript mailing list.

What are the parameters that `convert` uses wen calling ghostscript? Is there a way to check it?

**Comment by dlemstra** [comment]
If you add `-verbose` you can see the arguments.

**Comment by heitorPB** [comment]
update:

I [posted](https://ghostscript.com/pipermail/gs-devel/2020-September/010330.html) in the ghostscript mailing list and they fixed the bug. The fix will be available in gs 9.53.0.

Should I close this issue? Or wait until the release?

**Comment by heitorPB** [comment]
Ghostscript 9.53.0 was released today with the fix for this issue :)
